lib/src/views/misc/layout_switch.dart

```diff
@@ -0,0 +1,79 @@
+      width: double.infinity,
+      height: 50.0,
+      child: Stack(
+        alignment: Alignment.center,
+        children: [
+          GestureDetector(
+            onTap: () => ref.read(switchControllerProvider.notifier).onTap(SwitchPosition.left),
+            child: Padding(
+              padding: EdgeInsets.only(right: labelPadding),
+              child: Text(
+                context.i10n.labelList,
+                style: ref.watch(switchControllerProvider).position == SwitchPosition.left ? const TextStyle(fontWeight: FontWeight.w800) : null,
+              ),
+            ),
+          ),
+          GestureDetector(
+            onTap: () => ref.read(switchControllerProvider.notifier).onTap(),
+            onHorizontalDragStart: (_) => ref.read(switchControllerProvider.notifier).startDragging(),
+            onHorizontalDragUpdate: (DragUpdateDetails dragUpdateDetails) => ref.read(switchControllerProvider.notifier).handleDragging(dragUpdateDetails),
+            onHorizontalDragEnd: (_) => ref.read(switchControllerProvider.notifier).endDragging(),
+            child: Stack(
+              alignment: Alignment.center,
+              children: <Widget>[
+                Container(
+                  width: ref.watch(switchControllerProvider).trackWidth,
+                  height: 5.0,
+                  decoration: BoxDecoration(
+                    color: ref.watch(settingsControllerProvider).activeTheme.borderColor,
+                    borderRadius: BorderRadius.circular(25.0),
+                  ),
+                ),
+                AnimatedContainer(
+                  duration: ref.watch(switchControllerProvider).isDragging ? Duration.zero : const Duration(milliseconds: 150),
+                  curve: ref.watch(switchControllerProvider).isDragging ? Curves.linear : Curves.easeOut,
+                  transform: Matrix4.identity()..translate(ref.watch(switchControllerProvider).horizontalOffset, 0), // translate the dragged offset,
+                  child: Container(
+                    width: ref.watch(switchControllerProvider).knobWidth,
+                    height: ref.watch(switchControllerProvider).knobWidth,
+                    decoration: BoxDecoration(
+                      color: ref.watch(settingsControllerProvider).activeTheme.accentColor,
+                      shape: BoxShape.circle,
+                    ),
+                  ),
+                ),
+              ],
+            ),
+          ),
+          GestureDetector(
+            onTap: () => ref.read(switchControllerProvider.notifier).onTap(SwitchPosition.right),
+            child: Padding(
+              padding: EdgeInsets.only(left: labelPadding),
+              child: Text(
+                context.i10n.labelGrid,
+                style: ref.watch(switchControllerProvider).position == SwitchPosition.right ? const TextStyle(fontWeight: FontWeight.w800) : null,
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
```

Comment:
Very clean 😎 but I would add a comment above each `GestureDetector` perhaps so whoever viewing this code can more easily deduct what's what by quickly reading the comment instead of needing to go through the code.

Reply:
Added comments in 862d0889b77069f7b326e972904e9564870d8ceb.